### PR TITLE
Arcspc 658 for submission - title sort numeric-sensitive

### PR DESCRIPTION
**Pull request by michael-lts**
## Description
These requested changes fix an issue with title sorting in location and other tables in which numbers within alphanumeric title strings were sorting as strings. With these changes titles will sort ABC1, ABC2 ... instead of ABC1, ABC10 ... 
This fix requires a reindex.

- Changes to solr schema.xml: added icu_sort field type (now used by title_sort, identifier_sort, identifier instead of string_sort), which uses ICUCollationKeyFilterFactory with numeric sort; requires icu4j, solr-analysis-extras and lucene-analyzers-icu jars (note: the last 2 match solr version, e.g. lucene-analyzers-icu-4.10.4.jar). 

- Added above jars to ~/solr/lib.

- Changes to build.xml to support copying jars into solr.war. Additional changes to build.xml to add supporting jars to classpath during tests (temporarily copies jars to integration.tmp dir, then cleans up afterwards). 
(Note, the work done around adding and copying the supporting jars is only necessary for solr 4 in the war environment; in later versions of solr (non-war-based), the above jars are included with the solr distribution and just need to be referenced properly in the solrconfig.xml)

- One line change to indexer to strip multiple whitespace to 1 space in title sort field.

- New selenium test for sorting. 

## Related JIRA Ticket or GitHub Issue
#1853

## How Has This Been Tested?
Testing criteria: Numbers in titles should sort ABC1, ABC2 ... instead of ABC1, ABC10 ; if no numbers, sorting should not differ from previous sort.

Note: Testing to date done in local dev environment, and awaiting full testing in our QA environment.

Browse Location, sort by title:
/locations?sort=title_sort asc

Browse Accessions for a repository, sort by title:
/accessions?sort=title_sort asc

Same as Accessions for Resources, Digital Objects, Subjects, Agents and Container Profile

PUI (sort_title also used by PUI): sort Collections and Digital Materials by title

Ran full test suite including selenium

## Types of changes

- [ X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ x] I have read the **CONTRIBUTING** document.
- [x ] I have authority to submit this code.
- [x ] I have added tests to cover my changes.
- [ x] All new and existing tests passed.

### CODE REVIEW

**Comment by mark-cooper** [comment]
Hi @michael-lts 

@lorawoodford and I have started to look into this. Please let us know how your testing goes:

"Note: Testing to date done in local dev environment, and awaiting full testing in our QA environment."

Here are some things we'll be evaluatiing:

- Local dev testing with demo database and no data
- Local dev teting with MySQL and test database
- Local release build with MySQL and test database
- Local release build with MySQL and external Solr (version TBD)
- Remote test server with MySQL and test database

We'll mostly be focused on testing with Solr 4.10.4 (as the currently
supported version of Solr for aspace) but if you've made other changes
to your Solr config than identified in the PR let us know so we can take
that into account when testing with external Solr (you're using 6.x?).

Something we have been aiming to avoid is bundling large files in with
the core repo. The icu jar is 12mb so I'd suggest that one at least should
be downloaded as part of boostrap (and added to .gitignore).

File: build/build.xml [hunk 1]
```diff
@@ -352,7 +361,12 @@
   <target name="solr:war" depends="set-classpath, bootstrap-downloads" description="Repack the Solr .war file with our configs">
     <copy file="${solr_file}" tofile="solr.war" overwrite="true" />
     <war destfile="solr.war" update="true">
-      <zipfileset dir="../solr" prefix="WEB-INF/classes"/>
+      	    <zipfileset dir="../solr" includes="schema.xml" prefix="WEB-INF/classes"/>
+	    <zipfileset dir="../solr" includes="solrconfig.xml" prefix="WEB-INF/classes"/>
+	    <zipfileset dir="../solr" includes="stopwords.txt" prefix="WEB-INF/classes"/>
+	    <zipfileset dir="../solr" includes="synonyms.txt" prefix="WEB-INF/classes"/>
+   	    <zipfileset dir="../solr" includes="Dockerfile" prefix="WEB-INF/classes"/>
```
Comment (mark-cooper) [inline build/build.xml:368]: @michael-lts a small thing, the Dockerfile shouldn't be bundled with the war.

Comment (michael-lts) [reply build/build.xml:368]: Thank you @mark-cooper and @lorawoodford . We are still working on bringing up a shared test instance, and I will keep you posted on that qa testing. Thanks for letting me know about the eval plan, sounds good. There are additional solr fields in other pull requests, I will let you know about those soon. Your point about pulling the jars in bootstrap is a good one, I should have thought of it! And thank you for pointing out about not bundling the Dockerfile in the war.

Comment (michael-lts) [reply build/build.xml:368]: @mark-cooper, @lorawoodford, regarding new schema fields:
The additional new solr fields are in the schema on our developement branch here:
https://github.com/harvard-library/archivesspace/blob/development/solr/schema.xml
They are:
_u_abool (line 260)
_u_typeahead_utext (line 263)
_u_typeahead_usort (line 264 - uses new icu_sort)

Comment (michael-lts) [reply build/build.xml]: @mark-cooper and @lorawoodford , I made the suggested changes:
Added the 3 solr sort-related jars to bootstrap and removed from source
Added solr/lib jars to gitignore
Removed Dockerfile from solr:war target
I also added solr/lib/.gitkeep to initialize that directory for download
Thanks!

**Comment by mark-cooper** [comment]
@michael-lts thx for the updates. It looks like the test you added is unintentionally passing when run as an example and failing when the full suite of tests is run:

```txt
     [java] Failures:
     [java] 
     [java]   1) Locations Location batch correctly sorts locations in the browse list
     [java]      Failure/Error: expect(table_rows_location_text).to eq(table_rows_location_text_sorted)
     [java] 
     [java]        expected: ["1129 W. 81st St, 55, 66 MOO [Box XYZ: XYZ0001]", "1129 W. 81st St, 57, 67 MOO [Box ABC: ABC0001]", ...corner [Room: 1B, Shelf: 1]", "1978 Awesome Street, 6th, Studio 5, The corner [Room: 1B, Shelf: 2]"]
     [java]             got: ["123 Fake St, 13, 237, 37 [Box ABC: ABC0001]", "129 W. 81st St, 5, 5A [Box XYZ: XYZ0001]", "1129 W. ...corner [Room: 1B, Shelf: 1]", "1978 Awesome Street, 6th, Studio 5, The corner [Room: 1B, Shelf: 2]"]
```

To me it looks like the test comparison is inverted, I think it should be:

```ruby
expect(table_rows_location_text).not_to eq(table_rows_location_text_sorted)
# table_rows_location_text = numerically sorted from solr
# table_rows_location_text_sorted = string sorted by ruby
```

Then when all the tests are run:

```ruby
# this is the order from the table (table_rows_location_text)
table_rows_location_text = [
  "123 Fake St, 13, 237, 37 [Box ABC: ABC0001]",
  "129 W. 81st St, 5, 5A [Box XYZ: XYZ0001]",
  "555 Fake Street, 2nd, 201, Corner [Room: 1B, Shelf: 1]",
  "555 Fake Street, 2nd, 201, Corner [Room: 1B, Shelf: 2]",
  "555 Fake Street, 2nd, 201, Corner [Room: 1B, Shelf: 3]",
  "555 Fake Street, 2nd, 201, Corner [Room: 1B, Shelf: 4]",
  "1129 W. 81st St, 55, 66 MOO [Box XYZ: XYZ0001]",
  "1129 W. 81st St, 57, 67 MOO [Box ABC: ABC0001]",
  "1978 Awesome Street, 6th, Studio 5, The corner [Room: 1A, Shelf: 1]",
  "1978 Awesome Street, 6th, Studio 5, The corner [Room: 1A, Shelf: 2]"
]

# this is the comparison (table_rows_location_text.sort)
table_rows_location_text_sorted = [
  "1129 W. 81st St, 55, 66 MOO [Box XYZ: XYZ0001]",
  "1129 W. 81st St, 57, 67 MOO [Box ABC: ABC0001]",
  "123 Fake St, 13, 237, 37 [Box ABC: ABC0001]",
  "129 W. 81st St, 5, 5A [Box XYZ: XYZ0001]",
  "1978 Awesome Street, 6th, Studio 5, The corner [Room: 1A, Shelf: 1]",
  "1978 Awesome Street, 6th, Studio 5, The corner [Room: 1A, Shelf: 2]",
  "555 Fake Street, 2nd, 201, Corner [Room: 1B, Shelf: 1]",
  "555 Fake Street, 2nd, 201, Corner [Room: 1B, Shelf: 2]",
  "555 Fake Street, 2nd, 201, Corner [Room: 1B, Shelf: 3]",
  "555 Fake Street, 2nd, 201, Corner [Room: 1B, Shelf: 4]"
]
```

Here's a diff that will work with the full test suite and when using the specific example `./build/run frontend:selenium -Dexample='Locations'`:

https://gist.github.com/mark-cooper/300f49045304fe4431f0ad9f26177ac1

Feel free to change it up if you see a better approach to proving the sort order.

**Comment by michael-lts** [comment]
Hi @mark-cooper , thanks for the feedback and solution, I'll have a look at it here.

**Comment by mark-cooper** [comment]
@michael-lts I've completed the testing I mentioned in the 1st comment and haven't found any issues. Were you able to complete any testing on your end? Is there anything else you can think of that should be taken into consideration?

Before merging I'd like to see the direct references to "Harvard" removed as they shouldn't be retained in the core code.

**Comment by michael-lts** [comment]
Hi @mark-cooper , I just double checked, no new issues turned up during testing on our shared qa server. Let me look at the Harvard references

**Comment by michael-lts** [comment]
Hi @mark-cooper , I see the Harvard references in the schema.xml; along with removing the mentions of Harvard, should I also change 2020.1 to 1.6, or something else?

**Comment by mark-cooper** [comment]
@michael-lts I think it's ok to just remove the comments from schema.xml except for the last part:

```
<!-- below doesn't work with solr 4x - used old syntax above, switch to below for newer versions -->
<!-- <fieldType name="sort_icu" class="solr.ICUCollationField" locale="" strength="primary"
      numeric="true"/>-->
```

**Comment by michael-lts** [comment]
Hi @mark-cooper , I've removed those comments as requested.

**Pull request merged**